staging/src/k8s.io/component-base/zpages/flagz/flagz.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagz
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/munnerz/goautoneg"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	flagzHeaderFmt = `
+%s flags
+Warning: This endpoint is not meant to be machine parseable, has no formatting compatibility guarantees and is for debugging purposes only.
+
+`
+)
+
+var (
+	flagzSeparators         = []string{":", ": ", "=", " "}
+	acceptedMediaTypes      = []string{"text/plain"}
```

Comment:
we'll have more work to do in the future to wire up requests for json or something to the right handlers... can probably skip the abstraction for now and just have `acceptableMediaType` check directly for `"text"/"*"`, `"plain"/"*"`, empty or charset params, etc

Reply:
Got it, removed the variable for acceptedMediaTypes and hardcoded matching against "text/plain" for now.